File: 2320. Count Number of Ways to Place Houses/Solution.cpp
class Solution {
public:
    int countHousePlacements(int n)
    {
        int a = 1, b = 2, tmp;
        for (int i = 2; i <= n; ++i) {
            tmp = (a + b) % MOD;
            a = b;
            b = tmp;
        }
        return 1LL * b * b % MOD;
    }

private:
    constexpr static int MOD = 1000000007;
};

File: 2320. Count Number of Ways to Place Houses/Solution.py
class Solution:
    def countHousePlacements(self, n: int) -> int:
        MOD = 10 ** 9 + 7
        a, b = 1, 2
        for _ in range(2, n + 1):
            a, b = b, (a + b) % MOD
        return b * b % MOD
